lib/assert.js

```diff
@@ -281,12 +282,18 @@ function getErrMessage(message, fn) {
   overrideStackTrace.set(err, (_, stack) => stack);
   const call = err.stack[0];
 
-  const filename = call.getFileName();
+  let filename = call.getFileName();
   const line = call.getLineNumber() - 1;
   let column = call.getColumnNumber() - 1;
   let identifier;
   let code;
 
+  if (filename && StringPrototypeStartsWith(filename, 'file:')) {
+    try {
+      filename = fileURLToPath(filename);
```

Comment:
With the check for `file:`, under what conditions do we expect this to fail?

If we leave this `try/catch`, we might want to put a `debug` in the `catch`.

Reply:
``fileURLToPath`` and ``normalizeReferrerURL`` call ``URL`` constructor with a single argument. There was a string (I do not remember exactly, but I can try to remove the try/catch and rebuild to find it) like ``[eval something]`` that makes ``URL`` throws an ``ERR_INVALID_URL`` error. Since that is in the unit tests, it might be used in the future. Moreover, users of ``vm`` can pass any string for filename or have source map containing any string. The combination of both is possible to make otherwise invalid URL to appear as valid module and in general invalid URLs supplied by the user should not crush for the proper reason: ``MODULE_NOT_FOUND`` for example.

Reply:
Might be worth getting a unit test around the edge case if possible 👌